### PR TITLE
Icon conversion: support GraphicsMagick in addition to  ImageMagick

### PR DESCRIPTION
**Pull request by cmangla**
Work in progress. Comments welcome.

### CODE REVIEW

**Review by ronjouch — COMMENTED** [review]
Hi @cmangla . That looks good to me!

- Are you still working on this / is there anything to finish?
- Did you run a few smoke tests under macOS and Linux?
- In README.md and docs/API.md, can you update references to ImageMagick to "ImageMagick or GraphicsMagick" ?

**Comment by ronjouch** [comment]
Also! One more spot to patch: https://github.com/nativefier/nativefier/blob/0aa8276922b26f2fa7b026fcebb48bcf4a82ca48/src/helpers/helpers.ts#L71-L73

Please do an extra grepping pass of `convert` and `identify`, maybe I missed other usages.

**Comment by cmangla** [comment]
> Hi @cmangla . That looks good to me! Sorry for not commenting, I read "WIP", and mentally put your PR aside 😄.

No problem, thanks for your attention and comments.

> * Are you still working on this / is there anything to finish?

I could spend some time in order to finish this.

> * Did you run a few smoke tests under macOS and Linux?

I remember trying to test this on macOS. Could you point me to a simple test or example website I could use for this?

> * In README.md and docs/API.md, can you update references to ImageMagick to "ImageMagick or GraphicsMagick" ?

Will do.

**Comment by ronjouch** [comment]
> I remember trying to test this on macOS. Could you point me to a simple test or example website I could use for this?

@cmangla the basic test plan that comes to mind would be:

- Try under macOS with a website providing only a png icon, and check it's correctly converted to icns
- Try under Linux with a website providing only a ico icon, and check it's correctly converted to png
- Try under Linux with a website providing only a png icon, and check it's correctly used as untouched png

I don't have precise websites to test that, maybe I'd just spin up a local server with HTML I can fiddle with.

**Comment by cmangla** [comment]
I just went over this change, and realised that it will prioritise GraphicsMagic over ImageMagic if both are installed. Probably best to do the opposite, in order to match current behaviour.

**Comment by ronjouch** [comment]
> I just went over this change, and realised that it will prioritise GraphicsMagic over ImageMagic if both are installed. Probably best to do the opposite, in order to match current behaviour.

Your call, I wouldn't argue particularily about one over the other, or consider it a particularly breaking change.

**Comment by cmangla** [comment]
> I just went over this change, and realised that it will prioritise GraphicsMagic over ImageMagic if both are installed. Probably best to do the opposite, in order to match current behaviour.

Actually, I confused myself when I went over the code. The opposite is true, ImageMagic is prioritised over GraphicsMagic because it is the current default. Sorry, I started this quite a while ago.

File: README.md [hunk 1]
```diff
@@ -50,8 +50,8 @@ through the numerous open tabs when I was using [Facebook Messenger](https://mes
 - macOS 10.9+ / Windows / Linux
 - [Node.js](https://nodejs.org/) `>= 10` and npm `>= 6`
 - Optional dependencies:
-    - [ImageMagick](http://www.imagemagick.org/) to convert icons.
-      Make sure `convert` and `identify` are in your system `$PATH`.
+    - [ImageMagick](http://www.imagemagick.org/) or [GraphicsMagick](graphicsmagick.org) to convert icons.
```
Comment (ronjouch) [inline README.md:53]:
```suggestion
    - [ImageMagick](http://www.imagemagick.org/) or [GraphicsMagick](http://www.graphicsmagick.org/) to convert icons.
```

File: docs/api.md [hunk 2]
```diff
@@ -218,7 +218,7 @@ The icon parameter should be a path to a `.png` file.
 
 The icon parameter can either be a `.icns` or a `.png` file if the [optional dependencies](../README.md#optional-dependencies) are installed.
 
-If you have the optional dependencies `iconutil`, Imagemagick `convert`, and Imagemagick `identify` in your `PATH`, Nativefier will automatically convert the `.png` to a `.icns` for you.
+If you have the optional dependencies - `iconutil`; and Imagemagick `convert` and `identify`, or GraphicsMagic `gm` - in your `PATH`, Nativefier will automatically convert the `.png` to a `.icns` for you.
```
Comment (ronjouch) [inline docs/api.md:221]:
```suggestion
If your `PATH` has our image-conversion dependencies (`iconutil`, and either ImageMagick `convert` + `identify`, or GraphicsMagick `gm`), Nativefier will automatically convert the `.png` to a `.icns` for you.
```

**Review by ronjouch — APPROVED** [review]
Awesome, thanks again! Merging.

**Comment by ronjouch** [comment]
Released in v42.4.0. Thanks!

**Pull request merged**